libs/framework/doxygen.md

```diff
@@ -0,0 +1,66 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+   
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Intro
+
+The main way to use Celix is through the bundle context of a bundle.
+When a bundle is started the bundle context will be injected in the bundle activator.
+
+Apache Celix is a C framework with a C and C++ (C++11) API. 
```

Comment:
The reason why i used C++11 is because the C++ dependency manager stuff if C++11.
But yeah, this can be C++17 IMO.

Maybe this can be updated in a new PR, to prevent this PR from growing larger than it already is.

Reply:
I agree, another PR would be more suitable.